app/services/github_handler.rb

```diff
@@ -0,0 +1,64 @@
+class GithubHandler
+  def initialize(event, payload, signature)
+    @event = event
+    @payload = JSON.parse(payload)
+    @raw_payload = payload
+    @signature = signature
+  end
+
+  def handle
+    return unless webhook_verified?
+
+    case @event
+    when "pull_request"
+      case @payload["action"]
+      when "review_requested"
+        handle_review_request
+        # when "review_request_removed"
+        # TODO: Call the method to handle a review request removal
+        # else unsupported
+      end
+      # else unsupported
+    end
+  end
+
+  def handle_review_request
+    owner = create_or_find_user(@payload["pull_request"]["user"])
+    pr = create_or_find_pr(@payload["pull_request"])
+    # Even if you select multiple reviewers at once, the webhook sends a post for every person selected
+    # we can assume it is going to be a single user obj "requested_reviewer"
+    participants = [create_or_find_user(@payload["requested_reviewer"])]
+    ReviewRequest.create(data: @payload, owner: owner, users: participants, pull_request: pr)
+  end
+
+  def webhook_verified?
+    digest = OpenSSL::HMAC.hexdigest("SHA1", ENV["GITHUB_ANALYZER_WEBHOOK_SECRET"], @raw_payload)
+    ActiveSupport::SecurityUtils.secure_compare("sha1=#{digest}", @signature)
+  end
+
+  def create_or_find_user(json)
+    User.create_or_find_by(
```

Comment:
if the insertion fails because of a constraint, then it will try to find by all of the attributes. What happens if the `github_id` already exists in the db but the record has a different `login` ?